BlackNectar/BlackNectar/SideMenuFilterViewController.swift

```diff
@@ -187,4 +194,71 @@ extension SideMenuFilterViewController {
 
     }
 
+    
+    func setUserPreferences() {
+        
+        defaultPreferences.set(distanceFilter, forKey: "distanceFilter")
+        defaultPreferences.set(isRestaurant, forKey: "isRestaurant")
+        defaultPreferences.set(isOpenNow, forKey: "isOpenNow")
+        defaultPreferences.set(isStore, forKey: "isStore")
+        
+    }
+    
+    func loadDefaults() {
+        
+        if defaultPreferences.double(forKey: "distanceFilter") > 0 {
+            
+            distanceFilter = defaultPreferences.double(forKey: "distanceFilter")
+            isRestaurant = defaultPreferences.bool(forKey: "isRestaurant")
+            isOpenNow = defaultPreferences.bool(forKey: "isOpenNow")
+            isStore = defaultPreferences.bool(forKey: "isStore")
+            
+            let roundedNumber = (round(defaultPreferences.double(forKey: "distanceFilter") * 100)/100)
+            
+            slider.value = Float(defaultPreferences.double(forKey: "distanceFilter"))
+            slideValueLabel.text = "\(roundedNumber)"
+            
+            switch isRestaurant || isOpenNow || isStore {
```

Comment:
This logic is not best handled by a `switch` statement, but rather by a series of `if` statements.

```
if isRestaurant {
doRestaurantStuff()
}

if isStore {
doStoreStuff()
}

if isOpenNow {
doOpenNowStuff()
}
```

That is much clearer and more concise than the current switch statement.